x-pack/plugins/serverless_search/public/application/components/api_key/api_key.tsx

```diff
@@ -49,7 +49,7 @@ export const ApiKeyPanel = ({ setClientApiKey }: { setClientApiKey: (value: stri
         <CreateApiKeyFlyout
           onClose={() => setIsFlyoutOpen(false)}
           setApiKey={saveApiKey}
-          username={userProfile.user.full_name || userProfile.user.username}
+          username={user?.full_name || user?.username || ''}
```

Comment:
This LGTM, from looking at the code we only use this `username` for display on the form so I think everything should still work even is `user === undefined` right?